### PR TITLE
Update snowflake-configs.md

### PR DESCRIPTION
**Pull request by djayatillake**

### CODE REVIEW

**Comment by clrcrl** [comment]
@drewbanin — assigning you to this one as I don't know this part of the codebase that well

**Comment by drewbanin** [comment]
hey @djayatillake - thanks for opening this PR! I did some more testing here to confirm -- I don't think this updated documentation is necessarily correct, but there's definitely some valuable information in here.

Here's the test I ran:

```
-- models/clustered.sql
{{ config(materialized='table', cluster_by="left(uuid, 3)") }}

select 'abc123' as uuid
```

When I run
```
$ dbt run -m clustered
```

I see the following in the logs:
```
create or replace transient table "ANALYTICS".dbt_dbanin.cluster  as (
  select * from(
    select 'abc123' as uuid
   ) order by (left(uuid, 3))
);
....
alter table "ANALYTICS".dbt_dbanin.cluster cluster by (left(uuid, 3));
````

So, even without using the `list` syntax to specify a `cluster_by` config, I _am_ seeing dbt add the clustering keys as intended. Can you advise on the type of error message you saw when you tried something like this in the past?

-------

I really like the part of this PR where you call out that expressions can be used as clustering keys. I'll drop some suggestions inline in the PR here.

File: website/docs/docs/building-a-dbt-project/building-models/snowflake-configs.md [hunk 1]
```diff
@@ -139,6 +139,26 @@ create or replace table my_database.my_schema.my_table as (
  alter table my_database.my_schema.my_table cluster by (session_start);"
 ```
 
+It is possible to use functions of fields to cluster by but this must be enclosed in square brackets as below, even if only for a function on one field:
```
Comment (drewbanin) [inline website/docs/docs/building-a-dbt-project/building-models/snowflake-configs.md]: Maybe we could change this to something like:

> It is possible to use expressions in the `cluster_by` config as shown below

**Comment by djayatillake** [comment]
> hey @djayatillake - thanks for opening this PR! I did some more testing here to confirm -- I don't think this updated documentation is necessarily correct, but there's definitely some valuable information in here.
> 
> Here's the test I ran:
> 
> ```
> -- models/clustered.sql
> {{ config(materialized='table', cluster_by="left(uuid, 3)") }}
> 
> select 'abc123' as uuid
> ```
> 
> When I run
> 
> ```
> $ dbt run -m clustered
> ```
> 
> I see the following in the logs:
> 
> ```
> create or replace transient table "ANALYTICS".dbt_dbanin.cluster  as (
>   select * from(
>     select 'abc123' as uuid
>    ) order by (left(uuid, 3))
> );
> ....
> alter table "ANALYTICS".dbt_dbanin.cluster cluster by (left(uuid, 3));
> ```
> 
> So, even without using the `list` syntax to specify a `cluster_by` config, I _am_ seeing dbt add the clustering keys as intended. Can you advise on the type of error message you saw when you tried something like this in the past?
> 

This is strange, I can't reproduce the error again.

File: website/docs/docs/building-a-dbt-project/building-models/snowflake-configs.md [hunk 2]
```diff
@@ -139,6 +139,26 @@ create or replace table my_database.my_schema.my_table as (
  alter table my_database.my_schema.my_table cluster by (session_start);"
 ```
 
+It is possible to use functions of fields to cluster by, as below:
```
Comment (drewbanin) [inline website/docs/docs/building-a-dbt-project/building-models/snowflake-configs.md]: I think saying "expressions" might be clearer than "functions of fields" here - you buy that?

```suggestion
It is possible to cluster by expressions, as below:
```

**Review by drewbanin — COMMENTED** [review]
One last suggestion and then i think we'll be good to merge this :)

**Comment by VersusFacit** [comment]
@drewbanin Interested in having me "rescue" this PR. I'd have to do a bit of command line work to be able to see this through, but I am working through the repo to get us up to date.

Alternatively, if @djayatillake you're still with us, I'd love for you to do the honors of wrapping this up :)

**Comment by drewbanin** [comment]
blast from the past! Sure thing - if @djayatillake is down to revisit, would love to get this merged!

**Comment by djayatillake** [comment]
Sure let me take a look

**Comment by djayatillake** [comment]
I think it looks good now but there are conflicts to resolve

**Pull request closed without merging**